Fix boost.thread build with glibc >= 2.34, where PTHREAD_STACK_MIN is
defined as sysconf(_SC_THREAD_STACK_MIN) and can no longer be used in
preprocessor arithmetic. Backport of upstream boostorg/thread commit
74ff2db959c5fa75bec770c41ed2951a740fe936 (shipped in boost 1.78).

--- a/boost/thread/pthread/thread_data.hpp
+++ b/boost/thread/pthread/thread_data.hpp
@@ -57,7 +57,7 @@
 #else
           std::size_t page_size = ::sysconf( _SC_PAGESIZE);
 #endif
-#if PTHREAD_STACK_MIN > 0
+#ifdef PTHREAD_STACK_MIN
           if (size<PTHREAD_STACK_MIN) size=PTHREAD_STACK_MIN;
 #endif
           size = ((size+page_size-1)/page_size)*page_size;
